### ISSUE TITLE
Suggestion: Auto build with hammer or new wand.

### ISSUE THREAD

**Issue by Rated01010010** [Ipsis/Woot#84]
You should create controllers for each tier, and then when you right click the controller with the hammer, or a new wand/wrench, it auto builds the factory if you have all the blocks in your inventory. Similar to Environmental Tech multi-block structures. 

I will also be updating soon with some feedback to RF costs, when I get to testing out. Will probably open a separate issue, if you haven't made an official issue for discussing RF costs. 

Also, there seems to be a tier 4 factory, but it isn't in the wiki, and mobs like the Witch isn't working in it. Also in your block that displays how to build, there seems to be a typo. It goes Tier I, Tier II, Tier III, Tier III.

**Comment by Rated01010010** [comment]
Also, the controller should have a GUI, where you can insert and take out programmed mobs, instead of having to make a new controller for every mob and keeping them stored somewhere.

**Comment by Ipsis** [comment]
Unfortunately documentation is a developers weakness .... aside from the ability to texture.

There was an issue with the last release regarding lower cost mobs in a higher tier, which was fixed in 1.10.2-0.2.4-alpha. That issue was for Blaze, but would probably count for other mobs as well, or there is another bug.

I'll get the layout block display fixed, I can see in the code where the error is for that one.

**Comment by Ipsis** [comment]
Added bug 85 to handle the any power cost comments.

**Comment by Rated01010010** [comment]
Just some more spam heh

To the reports of crafting recipes being too hard, that can always be changed using mods like Crafting Harmonics. You should kepe it how you think the mod should be, and if they feel it's not balanced with their mod pack or server, then they can change it themselves.

I would however suggest allowing the mod pack creator/server owner to have full control in the config for things that cannot be changed using those type of mods, in determining RF costs and what mobs are available, etc.

Something like:
TierICostPerXLLevel: <#RF/tick cost>
TierIICostPerXLLevel: <#RF/tick cost>
TierIIICostPerXLLevel: <#RF/tick cost>
TierIVCostPerXLLevel: <#RF/tick cost>

MobNameEnabled: <true/false>
MobNameMinimumTierLevel: <1-4>
MobNameXPLevel: <1-?>

ModifierNameCost: <#RF/tick cost>

Your RF costs by default should be 25-50% higher than the standard mob spawners offered in other mods like Ender IO, as it is more of an "end game" feature for farming for mob drops, similar to void miners. Each higher tier should spawn more and quicker, but go up in RF cost per XP level. But they should not be lower than others, ever. The modifiers should cost substantially more. 

---------

I also suggest adding a GUI to the controller, so players can swap out what mob they want programmed, instead of creating more controllers.

And also the possibility of turning modifiers into items, that insert into the GUI instead, but that is no big deal.

--------

As for the issue of other mods spamming items on the ground, I don't think floating them up into the air is a reasonable fix. They should just be deleted. I'm sure there is someway in the Forge code to determine a player dropped entity to avoid those from being deleted.

**Comment by Ipsis** [comment]
You can blacklist or whitelist mobs for the prism to limit which mobs can be captured in the first place, and there are config options for mapping certain mobs to tiers (globalTierIMobList etc) and the base spawn cost (globalSpawnCostList). Plus there is the tier RF/tick (globalTierIRF etc).

I'm deliberately avoiding a GUI. I've programmed them before and I've never really been happy with them, which is basically why we have the upgrade totems.

For the mods spamming items, there are not very many options. I'm currently looking for dropped items a small area at the factory controller and basically deleting them as they are spawned. Once the  learning is finished for a mob/looting level, then those items stop being spawned. Once I've learned a mob/looting level pair, no mobs are spawned in any of the factories.

I'm still trying to find some way of preventing player dropped items from being picked up as well as I don't like the way I'm currently treating them. However it is down to knowing that a player dropped it in the first place. Of course there are always risks working beside heavy machinery .....

**Comment by Rated01010010** [comment]
True.

Also, Tier 2 mods are not working in Tier 4 factories it seems. Tier 1 and Tier 3 mobs work fine.

**Comment by Ipsis** [comment]
What version are you running, because I'm looking at the fix that should be in and now I'm wondering if it is still wrong?

**Comment by Rated01010010** [comment]
Well I'm chatting with a streamer on Beam, and he isn't getting it working. I told him you updated like ~4 hours ago. Was the fix in that version?

**Comment by Ipsis** [comment]
There is a fix in 1.10.2-0.2.4-alpha that should solve that. Of course anyone updating will need to remove their Woot config file as well, to make sure that they have the new RF setting.

**Comment by Rated01010010** [comment]
He's updating now I think

**Comment by Rated01010010** [comment]
So a few issues now.

When you break the mob controller to place another with a different programmed mob, it gets deleted. 

And also, he deleted the cow controller, which got deleted. And he placed a Blaze controller, and it's now making raw beef and leather. And after awhile it started making blaze rods. Very weird. 

Also with RF, does it work like other void miners, where once it has x RF in it's buffer, it performs a mob spawn, and then waits again? Or does it need a constant RF/tick? Because the latter way wouldn't keep up, unless there was a conduit that can support that rate.

**Comment by Ipsis** [comment]
Breaking the controller didn't delete it before. Wonder if something has changed in Forge that I didn't know about. I'm talking with Lumindia on IRC about the beef.

It needs a constant RF/tick, although it does have an RF buffer.

**Comment by Ipsis** [comment]
I know why it gets deleted ...... because the code that is capturing the spawning mob items that generate in the world .... it picking up the controller block as it falls.

So that issue just got pushed to the top of the list!
